pedsnetdcc/check_fact_relationship.py

```diff
@@ -0,0 +1,283 @@
+                raise RuntimeError('Data not returned from'
+                                   ' check_fact_relationship {0} query.'
+                                   .format(k))
+
+        if results['bad_fr_obs']['data'][0] != 0:
+            bad = results['bad_fr_obs']['data'][0]
+            total = results['tot_fr_obs']['data'][0]
+            percent = (bad * 100 // total)
+            logger.warn({
+                'msg': 'Fact relationship table has bad observation refs.',
+                'percent': percent,
+                'bad': bad,
+                'total': total
+            })
+        else:
+            logger.info({
+                'msg': 'Fact relationship observation refs are valid.'
+            })
+
+        if results['bad_fr_meas']['data'][0] != 0:
+            bad = results['bad_fr_meas']['data'][0]
+            total = results['tot_fr_meas']['data'][0]
+            percent = (bad * 100 // total)
+            logger.warn({
+                'msg': 'Fact relationship table has bad measurement refs.',
+                'percent': percent,
+                'bad': bad,
+                'total': total
+            })
+        else:
+            logger.info({
+                'msg': 'Fact relationship measurement refs are valid.'
+            })
+
+        if results['bad_fr_visit']['data'][0] != 0:
+            bad = results['bad_fr_visit']['data'][0]
+            total = results['tot_fr_visit']['data'][0]
+            percent = (bad * 100 // total)
+            logger.warn({
+                'msg': 'Fact relationship table has bad visit refs.',
+                'percent': percent,
+                'bad': bad,
+                'total': total
+            })
+        else:
+            logger.info({
+                'msg': 'Fact relationship visit refs are valid.'
+            })
+
+    elif output == 'samples':
+
+        sqls = {
+            'bad_fr_obs_id1_sample': sql_bad_obs_1_sample,
+            'bad_fr_obs_id2_sample': sql_bad_obs_2_sample,
+            'bad_fr_meas_id1_sample': sql_bad_meas_1_sample,
+            'bad_fr_meas_id2_sample': sql_bad_meas_2_sample,
+            'bad_fr_visit_id1_sample': sql_bad_visit_1_sample,
+            'bad_fr_visit_id2_sample': sql_bad_visit_2_sample
+        }
+
+        results = parallel_db_exec(conn_str, sqls, count=1)
+
+        for k, v in results.items():
+            if 'data' not in v:
+                logger.critical({'msg': 'Data not returned.', 'process': k})
+                raise RuntimeError('Data not returned from'
+                                   ' check_fact_relationship {0} query.'
+                                   .format(k))
+
+        if results['bad_fr_obs_id1_sample']['data']:
+            names = results['bad_fr_obs_id1_sample']['field_names']
+            data = results['bad_fr_obs_id1_sample']['data']
+            sample = dict(zip(names, data))
+            logger.warn({
+                'msg': 'Fact relationship table has bad observation refs.',
```

Comment:
Do we want to include the schema in the log messages? I guess the answer is no, given our typical workflows ...